net/FlatBuffers/Parser.cs

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2014 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace FlatBuffers
+{
+    public class Parser
+    {
+        private IntPtr m_parser;
+
+        public Parser()
+        {
+            m_parser = flatbuffers_parser_new();
+        }
+
+        public bool Parse(string source)
+        {
+            var ret = flatbuffers_parser_parse(m_parser, source);
+            if (ret == 0)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public bool SetRootType(string root_type)
+        {
+            var result = flatbuffers_parser_set_root_type(m_parser, root_type);
+            if (result == 0)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public string GenerateJson()
+        {
+            IntPtr buffer = new IntPtr();
+            IntPtr size = new IntPtr();
+
+            flatbuffers_generate_json(m_parser, ref buffer, ref size);
+            byte[] temp = new byte[size.ToInt32()];
+            Marshal.Copy(buffer, temp, 0, size.ToInt32());
+            var result = System.Text.Encoding.UTF8.GetString(temp);
+            if (size.ToInt32() > 0)
+            {
+                flatbuffers_free_string(buffer);
```

Comment:
Am i correct in understanding in that?
https://github.com/google/flatbuffers/blob/ac9041b7e96470f8ecb8ae833076dec0c1b90891/net/FlatBuffers/Parser.cs#L41

Currently, caller should handle exceptions when calling GeneratJson/GenerateBuffer method.